lib/languages/js/source-code/source-code.js

```diff
@@ -1002,16 +943,20 @@ class SourceCode extends TokenStore {
                 return false;
             }
 
-            const { directivePart } = commentParser.extractDirectiveComment(comment.value);
+            const directive = commentParser.parseDirective(comment.value);
+
+            if (!directive) {
+                return false;
+            }
 
-            const directiveMatch = directivesPattern.exec(directivePart);
+            const directiveMatch = directivesPattern.exec(directive.label);
 
             if (!directiveMatch) {
                 return false;
             }
```

Comment:
```suggestion
            if (!directivesPattern.test(directive.label)) {
                return false;
            }
```

As we now don't need captured text, it's more optimal to use `.test()`.